core/src/main/scala/org/broadinstitute/dsde/rawls/expressions/ExpressionValidator.scala

```diff
@@ -0,0 +1,50 @@
+package org.broadinstitute.dsde.rawls.expressions
+
+import org.broadinstitute.dsde.rawls.dataaccess.slick.DataAccess
+import org.broadinstitute.dsde.rawls.model.{AttributeString, MethodConfiguration, ValidatedMethodConfiguration}
+
+import scala.util.{Failure, Success, Try}
+
+import spray.json._
+
+/**
+  * Functions to help with the first phase of expression processing
+  *
+  * Phase 1: Validation.  Is this expression in the right format, given the context? e.g.
+  *   an empty string is fine for Output Expressions but not Input Expressions
+  *   JSON can be used to specify Input Expressions but not Output Expressions
+  *   *#@$ cannot be used in expressions
+  *   this..is.not.valid because it has empty elements
+  *
+  * Phase 2: Parsing.  Tokenize this (Validated) expression into a query path
```

Comment:
I think of it more like:

Validation - is this string allowed in the context of where it is in the MC? e.g. blank OK for optionals and outputs but not otherwise.

Parsing - is this string syntactically correct according to at least one parser? This would be where you reject `this..foo` for having two consecutive dots, or a bunch of nonsense symbols.

Evaluation - as you describe.

Right now you have calls to functions that begin with `parse` inside a thing that claims to be a validator, which seems off.

Reply:
OK.  So here it's a bit premature to call anything "Validator" since what we're doing is parsing.  For GAWB-2450 there will be a validator.